### ISSUE TITLE
Memory Error

### ISSUE THREAD

**Issue by SafakTanOzkan** [DaxStudio/DaxStudio#80]
Hello,
I get the following error;
"Memory error: Allocation failure. If using a 32-bit version of the product, consider upgrading to the 64-bit version or increasing the amount of memory available on the machine."

Hardware; Lenovo T560 Laptop i5 16 GB Ram 50 GB free Disk space for Virtual memory
Software: win10 64 bit, Office 2016 64 Bit, Dax studio 2.7.4;
Command: Evaluate Query1
target data: 75.747.214 rows of data from PowerPivot Query. ( Working fine in Excel though...)
 
All I could find on google is to increase Ram, switch from 32 to 64-bit suggestions which obviously is not applicable to my case, already 64 bit, plenty of RAM + Virtual (16 GB physical RAM)

**Comment by SafakTanOzkan** [comment]
Side note: I could pull many million rows of data at once with Dax Studio Before.
Not first time user. Great piece of software by the way. Thanks for your hard work :)

**Comment by dgosbell** [comment]
Off the top of my head that error message looks like it's coming from the PowerPivot engine, not from DAX Studio. The PowerPivot engine is specifically designed by Microsoft to work almost exclusively in physical RAM (it will not page out to disk)

What output target are you using? Am I correct in assuming that you are sending the output to a file, not to the output grid? When we output to a file we use a streaming API so the memory usage for DAX Studio should be pretty stable. Have you checked in Task Manager to see what process is using the most RAM when you run this query?

I'm not sure that you mean when you say that this query works in Excel - if you have 75 million rows there is no way to return that amount of data into an Excel sheet as they can only hold 1 million rows - are you just using a pivot table to report on summarize data?

**Comment by SafakTanOzkan** [comment]
I use both Grid and CSV output. The file is Heavy (520 MB) 75M rows in data model. you can download here;
https://drive.google.com/open?id=1fK08AaYl5wVfaRtjVgJI_1xv8_nBXcCo

**Comment by SafakTanOzkan** [comment]
My Dax Studio Checker results;

    Version =  v2.7.4.102

Checking Operating System
=======================
    OSCaption       = Microsoft Windows 10 Enterprise
    OSVersion       = 10.0.16299
    OSArchitecture  = 64-bit
    VisibleMemoryGB = 15,41
    FreeMemoryGB    = 11,68

    Culture Name              = tr-TR
    Culture DisplayName       = Turkish (Turkey)
    Culture EnglishName       = Turkish (Turkey)
    Culture 2-Letter ISO Name = tr
    Culture DecimalSeparator  = ,
    Culture GroupSeparator    = .
    Culture CurrencySymbol    = ₺
    Culture ShortDatePattern  = d.MM.yyyy

Checking .Net Framework
=======================
    v2.0.50727 -> 2.0.50727.4927
    v2.0.50727\1033 -> 2.0.50727.4927
    v3.0 -> 3.0.30729.4926
    v3.0\Setup -> 3.0.30729.4926
    v3.0\Setup\1033 -> 3.0.30729.4926
    v3.0\Setup\Windows Communication Foundation -> 3.0.4506.4926
    v3.0\Setup\Windows Presentation Foundation -> 3.0.6920.4902
    v3.5 -> 3.5.30729.4926
    v3.5\1033 -> 3.5.30729.4926
    v4\Client -> 4.7.02556
    v4\Client\1033 -> 4.7.02556
    v4\Full -> 4.7.02556
    v4\Full\1033 -> 4.7.02556
    v4.0\Client -> 4.0.0.0

Checking AMO
=======================
    FAIL > Could not load file or assembly 'Microsoft.AnalysisServices, Version=13.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
    PASS > Microsoft.AnalysisServices, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91
    PASS > Microsoft.AnalysisServices, Version=15.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91

Checking ADOMD.NET
=======================
    FAIL > Could not load file or assembly 'Microsoft.AnalysisServices.AdomdClient, Version=13.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
    PASS > Microsoft.AnalysisServices.AdomdClient, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91
    PASS > Microsoft.AnalysisServices.AdomdClient, Version=15.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91

Dax Studio Configuration
========================
    Path: "C:\Program Files\DAX Studio\DaxStudio.exe"
    Processing DaxStudio.exe.config file...
    Path: 'C:\Program Files\DAX Studio\DaxStudio.exe.config'
    Bindings Found: 9
    AMO : 14.0.0.0
    ADOMD : 14.0.0.0

Checking Excel Add-in
=======================
    Detected Excel Version: 16 - Excel_v16
    Excel Path: C:\Program Files\MIF5BA~1\Office16\EXCEL.EXE
    Excel Architecture: x64

    DAX Studio Excel Add-in Registry keys
       Description: Dax Studio Excel Add-In
       FriendlyName: Dax Studio Excel Add-In
       Manifest: file:///C:/Program Files/DAX Studio/DaxStudio.vsto|vstolocal
       LoadBehavior: 3

    Checking for Excel 2010 Disabled Add-ins
      PASS > No Disabled items found.
    Checking for Excel 2013 Disabled Add-ins
      PASS > No Disabled items found.
    Checking for Excel 2016 Disabled Add-ins
      PASS > No Disabled items found.

    Attempting to load Excel Add-in: 
      PASS > Loaded Excel Add-in: DaxStudio, Version=2.7.4.102, Culture=neutral, PublicKeyToken=null

    Checking VSTO Configuration
    Architecture: x64
      v4
        Version 10.0.60301
        Install 1
        InstallerPath C:\Program Files\Common Files\Microsoft Shared\VSTO\10.0\VSTOInstaller.exe
      v4M
        VSTOMFeature_UNM 1
        VSTOMFeature_CLR35 1
        VSTOMFeature_CLR40 1
      v4R
        VSTORFeature_UNM 1
        VSTORFeature_CLR35 1
        VSTORFeature_CLR40 1
        Version 10.0.60715
        Revision 00
        ProductCode {0630EB2F-AF6D-37E5-82D2-A3156D08E85B}

**Comment by dgosbell** [comment]
So we can ignore PowerQuery as you are only using that for loading data, you are not actually running the "EVALUATE Query1" query using DAX inside your PowerQuery so it's not the same thing. Loading data is done in 1M row segments and each segment is read in and compressed 1M rows at a time. However when you run an evaluate query you might be causing the whole table to be decompressed into a temporary buffer which could easily consume 10 or more times the size of your database.

What is it that you are trying to achieve? You already have the detailed information in a file, why are you loading it into a model then trying to read it back out again? 

Normally you'd load the data into a model so that you could run aggregate queries doing sums, counts, averages etc,

**Comment by SafakTanOzkan** [comment]
I use PowerQuery to merge multiple files ( txt, csv or xlxs) that have the same format. they represent some machine readings over a period of time but there are more than 600 000 files. in order to be able to run analysis on them ı need to pull them together. The reason I use DAX Studio is to be able to get the data out of Data model ( or power query is you will). I could not find another way "that works" to pull data out of excel without diving into 1Milyon row chunks. And actually, DAX studio worked well on smaller size batches like 10 M rows as far as I can remember. ( I also linked Data to PowerBI and used R script to pull it to CSV but I failed somehow.. but that's not related to our discussion)
